invenio_rdm_records/resources/serializers/dublincore/schema.py

```diff
@@ -39,11 +40,31 @@ class DublinCoreSchema(BaseSchema):
     locations = fields.Method("get_locations")
     formats = fields.Method("get_formats")
 
+    def _transform_identifier(self, identifier, scheme, prefix_scheme=True):
+        """Transform the raw identifier into a URL.
+
+        If the identifier can't be transformed into a URL and ``prefix_scheme`` is set
+        to ``True``, the identifier will take the shape ``{scheme}:{identifier}``.
+        Otherwise, the original identifier will be returned.
+        """
+        result = idutils.to_url(identifier, scheme, url_scheme="https")
+        if not result:
+            if prefix_scheme and not identifier.startswith(scheme):
+                result = f"{scheme}:{identifier}"
+            else:
+                result = identifier
+
+        return result
+
     def get_identifiers(self, obj):
         """Get identifiers."""
         items = []
-        items.extend(i["identifier"] for i in obj["metadata"].get("identifiers", []))
-        items.extend(p["identifier"] for p in obj.get("pids", {}).values())
+
+        for id_ in obj["metadata"].get("identifiers", []):
```

Comment:
Does the order have any impact on the output? In general we try to give priority to `pids` since those fields have much more validation and logic rather than a simple format validation.  However, if the consumer of dc data does not have any implications due to order, then it is all the same. Otherwise, I'd inver the order (first pids, then identifiers).

I know it was potentially wrong before, but we can profit not and fix it hehe

Reply:
sure thing, done :smile: